### PR TITLE
Add failed gcse information

### PR DESCRIPTION
**Pull request by tomas-stefano**
## Context

If a candidate does not have a GCSE A to C (or 4 or above) in English, Maths and Science, we ask them some follow-up question to see whether they are retaking their GCSE, or if they have any other evidence of meeting the GCSE requirement.

We should update the Support interface to surface this information.

### CODE REVIEW

**Comment by tomas-stefano** [comment]
[Surface additional GCSE information in the support interface](https://trello.com/c/wjaFJo7C/770-surface-additional-gcse-information-in-the-support-interface)

**Review by carlosmartinez — APPROVED** [review]

**Comment by frankieroberto** [comment]
@tomas-stefano just tested this in the review app, and this is how it appears:

Screenshot 2022-11-23 at 09 41 42

Looks good good me. One minor issue is that the "Other evidence I have the skills required" question should only be shown if the candidate does not have the GCSE C or above _and_ is not currently retaking the qualification. This is because if they ARE currently retaking a GCSE, they're not asked the question:

Screenshot 2022-11-23 at 09 42 20

If a candidate says they don't have the GCSE or equivalent qualification, but are currently studying for one, then I think instead of "Reason given" the label should be something like "Details of qualification currently studying for" - to match the question asked on the candidate side:

Screenshot 2022-11-23 at 09 47 09

If a candidate says they don't have the GCSE or equivalent qualification, and aren't currently studying for one, but gives evidence of meeting the skills, then I think the label should be "Other evidence of skills required" instead of "Reason given", to match the candidate interface and the other answers.

Happy to talk this through - and let's get this update on the provider interface too (if they don't share the same partials?).

File: app/components/shared/gcse_qualification_cards_component.html.erb [hunk 1]
```diff
@@ -11,7 +11,7 @@
             <h4 class="govuk-heading-s govuk-!-margin-bottom-1"><%= subject(qualification) %></h4>
             <dl class="app-qualification">
               <dd class="app-qualification__value"> <%= candidate_does_not_have %> </dd>
-              <dt class="app-qualification__key">Reason given</dt>
+              <dt class="app-qualification__key">Details of qualification currently studying for</dt>
               <dd class="app-qualification__value"><%= qualification.not_completed_explanation || qualification.missing_explanation %></dd>
```
Comment (frankieroberto) [inline app/components/shared/gcse_qualification_cards_component.html.erb:15]: This label needs to be conditional upon whether the candidate answered "Yes" or "No" to "Are you currently studying for this qualification?"

If they answered Yes, and then completed the free text, then the label should be "Details of qualification currently studying for". If they answered No, and then fill out the "If you have other evidence of having maths skills at the required standard, give details (optional)" then we should display "Other evidence I have the skills required" with their answer.

Currently this field can be mis-labelled:

Screenshot 2022-11-24 at 10 31 48

Hope that makes sense?

File: spec/components/utility/gcse_qualification_cards_component_spec.rb [hunk 2]
```diff
@@ -47,6 +47,28 @@
       end
     end
 
+    context 'when failed required gcse' do
```
Comment (frankieroberto) [inline spec/components/utility/gcse_qualification_cards_component_spec.rb:50]: Small tweak just to make clear that "D" isn't a "fail", but just below the ITT standard.

```suggestion
    context 'when their GCSE grade is below C or 4' do
```

**Review by frankieroberto — CHANGES_REQUESTED** [review]

**Review by frankieroberto — CHANGES_REQUESTED** [review]
Re-tested in review app.

Looks good. One last change. Where the candidate does not have the qualification and is not currently studying for it, and does not enter anything in the "Other evidence..." box (ie it's blank) can we add the text "Not provided"? Otherwise it looks a bit odd:

Screenshot 2022-11-30 at 11 46 25

**Review by frankieroberto — APPROVED** [review]

**Pull request merged**